Configure.cmake

```diff
@@ -0,0 +1,135 @@
+include(CheckCCompilerFlag)
+include(CheckCSourceCompiles)
+include(CheckTypeSize)
+
+# PLATFORM DETECTION
+
+if(WIN32)
+  set(_TMP_OSTYPE "MSVC")
+elseif(MSYS OR MINGW OR CYGWIN)
+  set(_TMP_OSTYPE "MinGW")
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  set(_TMP_OSTYPE "Linux")
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  set(_TMP_OSTYPE "Darwin")
+endif()
+
+# Cache OS type into a sleef variable and define it in the configuration header
+set(SLEEF_OSTYPE ${_TMP_OSTYPE} CACHE INTERNAL
+  "The OS 'type' for which SLEEF is configured.")
+
+# Cache machine architecture determined by checking size of a pointer
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+  set(SLEEF_ARCH_64BIT ON CACHE INTERNAL
+    "True for 64-bit architecture machines.")
+elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
+  set(SLEEF_ARCH_32BIT ON CACHE INTERNAL
```

Comment:
When it is compiled for 32 bit x86, we need -mfpmath=sse option. There is no harm to always add this option, though.